packit/utils.py

```diff
@@ -105,16 +105,19 @@ def run_command(
     :param env: set these env vars in the subprocess
     :param decode: decode stdout from utf8 to string
     :param print_live: print output from the command realtime to INFO log
+    :param shell: invoke in shell? (aka bash -c $cmd)
     """
-    if not isinstance(cmd, list):
-        cmd = shlex.split(cmd)
+    if shell:  # we are running a user defined action
+        # Popen( accepts cmd as a string, not list when shell=True - freaky
+        if isinstance(cmd, list):
+            cmd = " ".join(cmd)
+        cmd_str = cmd
+    else:  # command is defined by us and it's a List[str]
+        cmd_str = " ".join(cmd)
```

Comment:
Isn't it `cmd: Union[List[str], str]`?

We will get `c o m a n d` for `cmd='comand'`.

Reply:
show me a test case :P all are passing

in all seriousness, exactly what the comment says: when we define the command in our code, it's always `List[Str]` and from actions we are getting `Str` or a list of commands

Reply:
> and from actions we are getting Str

And it will call bad command in that case. (Like the example above, it will call `join` on a string and put a space between characters.)

(We have `cmd: Union[List[str], str]` now in the function header.)

Either support both or only `List[str]`, but type-annotation and implementation for all execution branches should match.

I got that problem once and it's very tricky to find the problem. (It is not a type error.)

Reply:
Ok, my bad. I combined `cmd` and `cmd_str`, but still. Comment is wrong and debug message will not make sense for `str` input.

Reply:
now I'm really lost, let's discuss in person in the tea house :) thanks for pointing this out